### ISSUE TITLE
add settings page

### ISSUE THREAD

**Issue by jonfinley** [seanvree/logarr#17]
Would be nice to be able to add and adjust settings on a settings page vs editing config.php

**Comment by seanvree** [comment]
Test build currently active in "settings" branch.  Having issue with PHP fwrite

**Comment by seanvree** [comment]
workflow moved to top comment.

**Comment by rob1998** [comment]
@jonfinley @seanvree should I add an option for live searching? So that it searches when something is being typed in the searchbox. Or do we not want it at all? (don't think performance will be very good with a lot of logs/lines....)

**Comment by jonfinley** [comment]
We had initially thought od doing that, but found the performance suffered greatly.

**Comment by seanvree** [comment]
@rob1998  yeah, you can test it out.  I think it will prob produce a memory leak, but you're welcome to try.  I would test with a huge ammount of logs...at least 10 logs with at least 3MB each and max limit of above 3000.

**Comment by seanvree** [comment]
@rob1998 @jonfinley  I'd like to add a changelog and/or application runtime log to the info page.  

There's a few repos that do this. Tested it for a min, Pretty easy, but here's the repo I like best:

https://github.com/conventional-changelog/standard-version/blob/master/CHANGELOG.md

Here's a couple others;
https://github.com/github-changelog-generator/github-changelog-generator



Here's a quick sketch of what I"m thinking:



![image](https://user-images.githubusercontent.com/8906637/41030025-36efc660-6932-11e8-928d-d718dbb16f43.png)

**Comment by rob1998** [comment]
I would do application changelog only.
Should be pretty easy to include that in the settings page, once you have it on github

**Comment by rob1998** [comment]
@seanvree I see you added custom error terms, do you want those to be log specific or general?
I think the easiest way to add them is to add a text field. Then separate terms by splitting everytime there's a comma in the string.

**Comment by seanvree** [comment]
Humm....I would probably say just general.  For example, highlight all "Warn" occurrences in yellow, same way we do "error"

**Comment by rob1998** [comment]
Okay, yeah so also no different colors?
Different colors would again require an array, so harder to implement in the settings page
All yellow is also good I think

**Comment by rob1998** [comment]
maybe we can include a class when content is being highlighted, for example warn. That way users can change colors using custom css. I think that's possible. (Just an idea for later)

**Comment by seanvree** [comment]
well, the best snecario would be something like Term: "error" color:  "Red"

**Comment by rob1998** [comment]
@seanvree yeah, but that would require an array (like the logs configuration), which doesn't look very nice in alpaca

Option 1:
I was thinking of a normal text input field where one would enter something like "error,warn,info". Then this string is being parsed by php/JS and every term is highlighted. Then the term is added as a class to each highlighted piece of text and with this you can edit the colors, we can add default classes for above terms (since they would be pretty standard I think)

Option 2:
Almost the same, but now with a string like this: "error:red,warn:yellow,info:blue", then the parsing would be a bit harder, but we could make it into an array or something that has the term as key and the color as value for example

Option 3:
The alpaca array field, really not a fan of this one. You saw what happened with the disk thingy

**Comment by seanvree** [comment]
@rob1998  option 3) we wouldn't have do an array if we do it for ALL logs right?

**Comment by seanvree** [comment]
@rob1998  I"m making notes to the first comment/checklist. Feel free to edit.

**Comment by rob1998** [comment]
@seanvree 
bugs:
- jump on search should be functional again, I think?

language:
it is possible to implement this, that's why I added it in. I could easily make something for that.
But again it's gonna take quite some time to implement it, basically replacing every string we have atm.

functionality:
- if you don't want a certain category you just shouldn't fill it in, when left blank there won't be a link for it. It has no other function than displaying that link atm. Categories are "automatically" created if a "new" one is found in one of the logs data. I see you wanted to add the category to the log container, but I don't know how useful/necessary that would be.
- If you set the log roll size to 0, it won't roll it. So that's a toggle, right?
- Individual updating is something I've been thinking about, I was thinking of saving it for the next release, since it's kind of a big update again and I just wanted to be sure that this all works

style:
- I don't remember atm why I made the links on the login/register page, but they were not only for dev purpose, so I should check that. Register functionality is something we should still discuss, gonna save that discussion for when I'm back. The links might have to do with the register functionality
- I didn't know what to do with the tools option, I put it in, in case I would find something that we could use as tool, but couldn't think of anything.
- I'm still not a fan of the title centered to the container instead of the page, but I guess I could add in my custom CSS again for myself. Be aware: the page title and footer are actually part of the page's html instead of the container html
- logo's: I would say on every place the logo is atm and on the settings page where it says Logarr (above the clock)
- what do you mean that you're not a fan of the log-on container? Just stylewise? Cause I just made the most basic login form I could think of.

**Comment by seanvree** [comment]
@rob1998 

- Jump on search:
I'm using chrome, and it always "jumps" despite the setting.  Shoot me your site so I can test and make sure it's not just me, also, testing on chrome, or something else?

- Language:  
Are you thinking client-side type of translation which would only translate for that browser session or an actual back-end full-text php-based translation solution? If the latter which library are you thinking? I know this a pretty big job.  We can leave it in the UI, just wondering. 

- Categories:
Yeah, I got the "functionality" of the categories, what I mean is what if you had three categories: 1, 2 & 3, but you wanted to ONLY display cat 1 and 2?  That's why I was thinking some type of toggle switch for each category that could be enabled/disabled independently of each other. I actually really liked this, so I did a lot of work on it and got it semi-functional on my site:

![image](https://user-images.githubusercontent.com/8906637/42989822-22ed2be0-8bbe-11e8-8edb-f2aa0d1e7946.png)

 The toggle switches are NOT functional, but you can see how it might look. I haven't uploaded the code because I wanted to see what you thought.  http://seanvree.com/dev/logarr-settings/
(username : password)

What do you guys think? @jonfinley  ?

- Tools option:  Yeah I can't think of anything, so unless we need it, let's remove it, and I would say add a "register" nav menu option for that?

- Title center:  I mean we took a vote buddy, so I hate to say it, but you might have to do it with your custom CSS. LOL

- Logos:  Yeah, I'll take a closer look it's obvious where it will go on the main log page - but as far as the settings pages, I think your right to put it above the clock. (@jonfinley  what's your vote on this?)

- Log-on container:  I mean, I'm not a fan of the grey container that encapsuates the username and password fields, I prefer the bottom option in the screenshot below:

![image](https://user-images.githubusercontent.com/8906637/42991238-3c53cdf6-8bc2-11e8-9c0e-597c0ad998c6.png)

**Comment by rob1998** [comment]
@seanvree 
okay so quick answer (will be researching and fixing stuff you mentioned in the upcoming days):

jump on search:
I'll have to check that, might be a bug

language:
I was thinking serverside, don't really need a library for it. But I'll look into it further

categories:
looks like a good idea, although I would use checkboxes instead of toggles?

tools option:
okay

title center:
haha yeah I know, I don't really care since I can use custom CSS now :wink:

log-on container:
ah okay, that's styling: I'll leave that up to you :wink:

**Comment by seanvree** [comment]
BUG:

add required extension:  

`extension=openssl`

**Comment by seanvree** [comment]
BUG add to functions.php:

`ini_set('error_reporting', E_ERROR);`

**Comment by seanvree** [comment]
add to PHPinfo.php:

extension check:

- extension=openssl
- write check (https://github.com/Monitorr/Monitorr/issues/200)

**Comment by seanvree** [comment]
BUG:

fix favicon.png in webmanifest.json

**Comment by seanvree** [comment]
CHANGE:

- Add test log for default log

**Comment by seanvree** [comment]
Add:

- BlockUI on settings change

**Comment by seanvree** [comment]
BUG:

- Add login requirement for all settings pages
